source/pipe.js

```diff
@@ -32,6 +35,12 @@ export default function pipe() {
   if (arguments.length === 0) {
     throw new Error('pipe requires at least one argument');
   }
+  const firstNonFunction = [].slice.call(arguments).find(complement(_isFunction));
```

Comment:
I wasn't sure whether I should be using ramda functions for this internal check, so I'd be happy to make this `ff => !_isFunction(ff)` if that's preferable.

Reply:
Those internal functions are there precisely to be used in places like this, so... yes!

Reply:
We can't use Arrow functions until we drop support of ES5. I think `complement` is OK here.

Reply:
For me at least I'd like to see it more similar to how we perform list iterations in general insead of using the built in Array functions.

what about this:
```
var args = Array.prototype.slice.call(arguments);
var idx = 0;
  while (idx < args.length) {
    if (!_isFunction(args[idx])) {
          throw new TypeError(
            'pipe: ' + toString(args[idx]) + ' is not a function'
    );
    }
    idx += 1;
  }
return blah;
```

@brekk @CrossEye @team?

Reply:
Updated.